runbot_travis2docker/models/runbot_branch.py

```diff
@@ -3,20 +3,87 @@
+                        dict(match.groupdict(), branch=branch['branch_name']))
+            branch.name_weblate = name
 
     @api.model
     def cron_weblate(self):
         for branch in self.search([('uses_weblate', '=', True)]):
-            self.env['runbot.build'].create({'branch_id': branch.id,
-                                             'name': 'HEAD',
-                                             'uses_weblate': True})
+            if (not branch.repo_id.weblate_token or
+                    not branch.repo_id.weblate_url):
+                continue
+            current_date = (datetime.strptime(branch.updated_weblate,
+                                              '%Y-%m-%d %H:%M:%S')
+                            if branch.updated_weblate
+                            else None)
+            new_date = None
+            url = branch.repo_id.weblate_url
+            session = requests.Session()
+            session.headers.update({
+                'Accept': 'application/json',
+                'User-Agent': 'runbot_travis2docker',
+                'Authorization': 'Token %s' % branch.repo_id.weblate_token
+            })
+            projects = session.get(url + '/projects/').json()
+            for project in projects['results']:
+                components = session.get('%s/projects/%s/components'
+                                         % (url, project['slug'])).json()
+                for component in components['results']:
+                    if component['branch'] != branch['branch_name']:
+                        continue
+                    if project['name'] != branch.name_weblate:
+                        continue
+                    changes = session.get('%s/components/%s/%s/changes/'
+                                          % (url, project['slug'],
+                                             component['slug'])).json()
+                    if not changes['results']:
+                        continue
+                    change = iter(changes['results']).next()
+                    if change['action'] != 17:
+                        continue
+                    date = datetime.strptime(
+                        change['timestamp'], '%Y-%m-%dT%H:%M:%S.%fZ')
+                    new_date = (date
+                                if (not new_date or date > new_date)
+                                else new_date)
+                    if ((current_date and new_date and
+                            current_date < new_date.replace(microsecond=0)) or
+                            (not current_date and new_date)):
+                        branch.write({'updated_weblate':
+                                      new_date.strftime('%Y-%m-%d %H:%M:%S')})
+                        self.env['runbot.build'].create({
+                            'branch_id': branch.id,
+                            'name': branch.branch_name,
+                            'uses_weblate': True})
+                        break
```

Comment:
I'll merge it but I think that we need the break out of the `if`
I mean, if the branch was found it so is not required search again